### ISSUE TITLE
Max Hit Magic not showing correctly.

### ISSUE THREAD

**Issue by BeeZZy03** [j-cob44/max-hit-calc#41]
ill add a photo of gear, just in case its my tome of water or something thats messing it up.
![Screenshot (15)](https://github.com/user-attachments/assets/76432c89-19ba-4bce-bf9e-2449288c546d)

Equipment:
imbued slayer helmet
Saradomin Cape
Occult Necklace
Peaceful Blessing
Ahrim's Staff
Ahrims's Robetop
Tome of Water
Ahrim's Robeskirt
Tormented Bracelet
Inifnity Boots
Seers Ring

**Comment by BeeZZy03** [comment]
Water Wave level 75 magic

**Comment by BeeZZy03** [comment]
The baby blue dragons have 50% water elemental weakness. I’m assuming the max hit wasn’t taking this into account.

**Comment by BeeZZy03** [comment]
![IMG_1536](https://github.com/user-attachments/assets/fe9cf0a5-48ce-4ead-9549-499779e8b476)

**Comment by j-cob44** [comment]
It should be taken into account, but it might not be. I've been busy these past few days so I haven't had a chance to test it yet. Just a little busy, but I appreciate you diagnosing it further.

**Comment by BeeZZy03** [comment]
Well it’s not on brutal blue dragons. I’m maxing 41  but it says 28

**Comment by j-cob44** [comment]
Hey @BeeZZy03, I've finally had a chance to finish this up. I've fixed this and it will be included in the v2.0.2. 

The problem was that the spell type weakness was not being correctly applied to NPCs with names deriving from another name. For example, the baby blue dragon spell weakness is the same as the normal blue dragon and is supposed to use the blue dragon's information. This wasn't working properly and is now fixed.
